onnx/backend/test/case/node/negativeloglikelihoodloss.py

```diff
@@ -11,74 +11,60 @@
 
 
 def compute_negative_log_likelihood_loss(input, target, weight=None, reduction='mean', ignore_index=None):  # type: ignore
-    ''' Compute negative_log_likelihood_loss '''
     input_shape = input.shape
+    target_shape = target.shape
+    N = input_shape[0]
+    C = input_shape[1]
+
+    # initialize the positional weights when required
+    gather_weight = None
+    if weight is not None:
+        # setting mode='clip' to deal with ignore_index > C cases.
+        # when the target valeu is > C, it doesn't matter which value we are
+        # taking in gather_weight, since it will be set to 0 in the following lines
+        gather_weight = np.take(weight, target, mode='clip')
+        if ignore_index is not None:
+            gather_weight = np.where(target == ignore_index, 0, gather_weight).astype(dtype=np.float32)
+    elif ignore_index is not None:
+        gather_weight = np.where(target == ignore_index, 0, 1).astype(dtype=np.float32)
+
+    # if input is 4-d and above, make it 3-d
+    if len(input_shape) > 3:
```

Comment:
We can reshape input shape `[N, C]` to `[N, C, 1]` and then all the following code become applicable to all cases without if-else. Note that we still need to throw if the input rank is less than 2.